src/main/java/io/cdap/plugin/gcp/common/GCPReferenceSourceConfig.java

```diff
@@ -29,17 +29,17 @@
 public class GCPReferenceSourceConfig extends GCPConfig {
   @Name(Constants.Reference.REFERENCE_NAME)
   @Description("This will be used to uniquely identify this source for lineage, annotating metadata, etc.")
-  @Macro
   public String referenceName;
 
   /**
    * Validates the given referenceName to consists of characters allowed to represent a dataset.
    */
-  public void validate() {
+  public void validate(FailureCollector collector) {
     try {
       IdUtils.validateId(referenceName);
     } catch (IllegalArgumentException e) {
-      throw new InvalidConfigPropertyException(e.getMessage(), Constants.Reference.REFERENCE_NAME);
+      collector.addFailure(e.getMessage(), "Supported characters are: letters, numbers, and _, -, ., or $.")
```

Comment:
probably better to modify IdUtils

Reply:
yes, Will add it there. That would also mean that existing `IdUtils#validateId` should be deprecated.

Reply:
Fixed: https://github.com/cdapio/hydrator-plugins/pull/935